low-level/files/opt/seagate/sspl/bin/manifest_support_bundle

```diff
@@ -23,213 +23,230 @@ import sys
-                     "drive-serial-number"]
-node_summary_fields=['id', 'class', 'description', 'product', 'serial', 'vendor',
-                     'part-number', 'model-number','children', 'physid', 'version']
-
-def make_manifest_support_bundle_tarfile(parser, tmpFile):
-    configure_path = parser[2]
-    if configure_path:
-        if os.path.exists(configure_path):
-            localTempPath = configure_path+"manifest/"
-        else:
-            print("Given path doesn't exist")
-            tmpFile.close()
-            os.remove(onGoingRequestPath)
-            sys.exit(1)
-    if not os.path.exists(localTempPath):
-        os.makedirs(localTempPath)
-    try:
-        sspl_state = ""
+
+class ManifestSupportBundle(object):
+
+    def __init__(self):
+
+        self.localTempPath = "/tmp/support_bundle/"
+        self.encl_manifest = "/tmp/encl_manifest.json"
+        self.node_manifest = "/tmp/node_manifest.json"
+        self.manifest_summary = "/tmp/manifest_summary.json"
+        self.lr_serial_number_path = "/opt/seagate/lr-serial-number"
+        self.node_hostname = socket.gethostname().split('.', 1)[0]
+        self.encl_summary_filter={}
+        cluster_id = subprocess.Popen(['salt-call', 'grains.get', 'cluster_id'], stdout=subprocess.PIPE).communicate()[0]
+        self.cluster_id = cluster_id.decode("utf-8").rstrip().replace('local:\n    ', '')
+        self.encl_summary_fields = ["object-name", "durable-id", "vendor-name", "product-id", "serial-number",
+                            "model", "product-brand", "description", "status", "health", "health-reason",
+                            "platform-type", "platform-brand", "part-number", "enclosure-wwn", "vendor",
+                            "drive-serial-number"]
+        self.node_summary_fields = ['id', 'class', 'description', 'product', 'serial', 'vendor',
+                            'part-number', 'model-number','children', 'physid', 'version']
+
+
+    def make_manifest_support_bundle_tarfile(self, parser, tmpFile):
+        configure_path = parser[2]
+        if configure_path:
+            if os.path.exists(configure_path):
+                self.localTempPath = configure_path+"manifest/"
+            else:
+                print("Given path doesn't exist")
+                tmpFile.close()
+                os.remove(onGoingRequestPath)
+                sys.exit(1)
+        if not os.path.exists(self.localTempPath):
+            os.makedirs(self.localTempPath)
         try:
-            sspl_state=open(f"/var/{PRODUCT_FAMILY}/sspl/data/state.txt").readline().rstrip().split("=")
-            # Capturing enclosure data only when sspl state is active
-            sspl_state = sspl_state[1] if len(sspl_state)==2 else ''
-            if sspl_state == 'active':
+            sspl_state = ""
+            try:
+                lr_serial_number = open(self.lr_serial_number_path).readline().rstrip()
```

Comment:
this is the only change which we seeked through this patch majorly

Reply:
Yes